### DISCUSSION TITLE
Unwanted merge commits in PR

### DISCUSSION THREAD

**Discussion by THATotaku** [community/community#22397] [New to GitHub]
<p>Shown by the image below, in my latest PR (and the one before), there have been 4 commits instead of the intended 1 (the last one (611f1e0) is the only one I wanted). The other 3 commits seem to be from weeks ago. Its quite annoying and messy, and haven’t been able to find a way to remove them after some decent searching.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/f/6/f661aea2670577cb4fe2d1527fa439cf1a128047.png" data-download-href="/uploads/short-url/z9AJVEznkwjujE7RoLcTSuwAPCD.png?dl=1" title="reeeeeeeee" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/82127230/181070750-52ec06fa-7e10-411f-afaa-14dbdd4f9704.png" alt="reeeeeeeee" data-base62-sha1="z9AJVEznkwjujE7RoLcTSuwAPCD" width="690" height="384" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/f/6/f661aea2670577cb4fe2d1527fa439cf1a128047_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">reeeeeeeee</span><span class="informations">949×529 25 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Does anyone know how to get rid of the unwanted commits? (And is there any reason not to?) Thank you :slight_smile:</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Those commits look like you merged updates from the branch that you’re trying to merge into. If you want to avoid that, the best solution is probably to start a fresh branch from the current state of the upstream branch, and <a href="https://git-scm.com/docs/git-cherry-pick"><code>cherry-pick</code></a> your commit on top of that.</p>
<p>If you want to avoid such merge commits in the future, you can <a href="https://git-scm.com/docs/git-rebase">rebase</a> your branch on top of upstream changes instead of merging them.</p>

**Comment by THATotaku** [comment]
<p>Thanks for the help! will give it a go tonight :’)</p>
